ui/src/components/TableEntity.js

```diff
@@ -2,36 +2,49 @@ import React from 'react';
 import { css } from '@emotion/react';
 import { stringify } from 'query-string';
 
-const TableEntity = ({ sqon, savedSetsContext, state, value, history }) => (
-  <button
-    className="clickable"
-    css={css`
-      background: none;
-      border: none;
-    `}
-    onClick={async () => {
-      const { setId } = await savedSetsContext.createSet({
-        sqon,
-        sort: [...state.sorted, { id: 'name', desc: false }].map(({ id, desc }) => ({
-          field: id,
-          order: desc ? 'desc' : 'asc',
-        })),
-      });
-      if (setId) {
-        history.push({
-          pathname: `/model/${value}`,
-          search: stringify({
-            sqon: JSON.stringify({
-              op: 'in',
-              content: { fieldName: 'setId', value: setId },
+const TableEntity = ({
+  sqon,
+  savedSetsContext,
+  searchWrapperSetState,
+  tableState,
+  value,
+  history,
+}) => {
+  const { createSet } = savedSetsContext;
+  return (
+    <button
+      className="clickable"
+      css={css`
+        background: none;
+        border: none;
+      `}
+      onClick={async () => {
+        searchWrapperSetState({ loading: true });
```

Comment:
Should we manually set loading back to `false` after? I guess we're navigating the user to `/model/:modelId` after, but that's conditional on successfully creating a set.

Could be overkill, but how would you feel about setting loading back to `false` after that `if (setId)` conditional, _just in case_ something goes wrong creating the set?